src/make_watertight/Gen.hpp

```diff
@@ -0,0 +1,279 @@
+#ifndef GEN_HPP
+#define GEN_HPP
+
+#include <iostream>
+#include <iomanip> // for setprecision
+#include <limits> // for min/max values
+#include <assert.h>
+#include <math.h>
+#include <time.h>
+#include <vector>
+#include <algorithm>
+#include "moab/Core.hpp"
+#include "MBTagConventions.hpp"
+#include "moab/Range.hpp"
+#include "moab/AdaptiveKDTree.hpp" // for merging verts
+#include "moab/CartVect.hpp"
+
+class Gen {
+ public:
+  Gen(moab::Interface* mbInterface) : mbi(mbInterface) {};
+
+
+  ~Gen() {};
+
+  moab::Interface* mbi;
+  moab::Interface* MBI() {
+    return mbi;
+  };
+
+
+  /// prints a string of the error code returned from MOAB to standard output
+  void moab_printer(moab::ErrorCode error_code);
+
+  void print_vertex_cubit(const moab::EntityHandle vertex);
+
+  void print_arc_of_edges(const std::vector<moab::EntityHandle> arc_of_edges);
+
+  void print_vertex_coords(const moab::EntityHandle vertex);
+
+  void print_triangles(const moab::Range tris);
+
+  void print_triangle(const moab::EntityHandle triangle, bool print_edges);
+
+  void print_edge(const moab::EntityHandle edge);
+
+  void print_vertex_count(const moab::EntityHandle input_meshset);
+
+  void print_range(const moab::Range range);
+
+  void print_range_of_edges(const moab::Range range);
+
+  void print_arcs(const std::vector < std::vector<moab::EntityHandle> > arcs);
+
+  void print_loop(const std::vector<moab::EntityHandle> loop_of_verts);
+
+  moab::ErrorCode find_closest_vert(const moab::EntityHandle reference_vert,
+                                    const std::vector<moab::EntityHandle> arc_of_verts,
```

Comment:
Efficiency: When passing around large objects such as vectors that the method does not modify (`const`), you can optionally use a `&` to pass by reference.  When combined with `const`, passing by reference provides the same data to the method without having to make a copy of the whole vector [1].

[1] https://stackoverflow.com/a/57746586/1015951